### ISSUE TITLE
EfficientNet Edge Small ImageNet accuracy on Coral USB

### ISSUE THREAD

**Issue by volcacius** [google-coral/edgetpu#199]
Hello,

I'm trying to reproduce the top1 accuracy on ImageNet validation reported [here](https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet/edgetpu) for the quantized EfficientNet Edge Small distributed on the Coral website ([link to .tflite](https://github.com/google-coral/edgetpu/raw/master/test_data/efficientnet-edgetpu-S_quant_edgetpu.tflite)), running on the Coral USB accelerator. 
In my current setup the measured accuracy stays below 60% top1 accuracy (it should be 77%), and I can't spot what the issue might be, so I would appreciate any help with this. 
The same setup can reproduce accuracy (above 70%) on the quantized MobileNet V1 ([link to .tflite](https://github.com/google-coral/edgetpu/raw/master/test_data/mobilenet_v1_1.0_224_quant_edgetpu.tflite)).

A documented approach to reproduced ImageNet validation accuracy with Tensorflow Lite is to compile this [script](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/tools/evaluation/tasks/imagenet_image_classification), however the EdgeTPU doesn't seem to be supported as a delegate runtime, so I wrote my own small python script based on official examples.

I attached a zip containing the following:

- _model_labels.txt_ - 1001 labels generated by the model
- _generated_validation_labels.txt_ - 50000 labels corresponding to the ImageNet validation data, generated with[ this](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/tools/evaluation/tasks/imagenet_image_classification/generate_validation_labels.py).
- _imagenet_evaluate.py_ - Python script to compute top1 accuracy. Pillow and the tflite runtime are the only requirements.

You can invoke the script as:
`python -m path/to/edgetpu_model.tflite -i path/to/imagenet/validation/folder -v path/to/generated_validation_labels.txt -l path/to/model_labels.txt`

The ImageNet validation folder is expected to contains file named ILSVRC2012_val_INDEX, with _INDEX_ the numbers between 00000001 and 00500000.

My current setup running on Windows 10:

- edgetpu_runtime_20200728
- python 3.7.0
- pillow 7.2.0
- tflite-runtime 2.1.0.post1

Attachment: [debug_coral_imagenet_val.zip](https://github.com/google-coral/edgetpu/files/5074065/debug_coral_imagenet_val.zip)

Thanks

Alessandro

**Comment by Namburger** [comment]
@volcacius 
Thanks for reporting, I'm waiting for the imagenet's permission to download the validation data and I'll give this test a try. I can look at your source code to see if something is off once I can get the dataset.

To clarify, you are using the [provided model](https://github.com/google-coral/edgetpu/raw/master/test_data/efficientnet-edgetpu-S_quant_edgetpu.tflite), correct?
Since the model was create, trained, and evaluated by a different team, I suggest you also open an issue [here](https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet/edgetpu#using-pretrained-efficientnet-edgetpu-checkpoints) to find out how the results was produced!

Thanks

**Comment by volcacius** [comment]
Hello,

Yes I'm using the provided model linked. 
I just had a look and noticed the problem had already been reported by someone last year [here](https://github.com/tensorflow/tpu/issues/483). They suggest might be preprocessing, so I tried changing crop to 256x224 & resize to 224x224 to just resize to 224x224, but that doesn't seem to make any difference. So there was no other follow up, so I'll ask if there are any updates.

**Comment by Namburger** [comment]
@volcacius I see, I just took a look and ran your code, it looks good (very clean). I also tried our [preprocessing](https://github.com/google-coral/edgetpu/blob/master/edgetpu/utils/image_processing.py#L21) utils function but still get roughly the same results. A couple things I would try:

1) Use [this preprocessing](https://github.com/tensorflow/tpu/blob/master/models/official/efficientnet/export_model.py#L70) algo to make sure it is an exact match of when training/post training quantization was done.
2) Try re-exporting the checkpoint and make your own tflite model?

**Comment by volcacius** [comment]
@Namburger thanks for the reply, sorry I have been off work. 

I'll try (1) and report back. With (2), do you re-doing the calibration step to quantize from float to int8?

In general though it would be really helpful if there was some official indication from Google of how to properly reproduce the results, especially given that EfficientNet Edge was supposed to be _the_ model designed to run on the EdgeTPU. They are mentioned and linked across multiple websites, but at the moment it doesn't seem like (the pretrained variants) are usable for anything.

Thanks
